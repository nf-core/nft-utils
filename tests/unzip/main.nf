workflow {

  ch_stable_content = Channel
    .of(
      """
    I HAVE STABLE CONTENT
    """.stripIndent().trim()
    )
    .collectFile(storeDir: "${params.outdir}/stable", name: 'stable_content.txt', sort: true, newLine: true)
}
