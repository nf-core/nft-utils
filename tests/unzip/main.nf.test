import nf_core.nf.test.utils.ZipUtilsWrapper

nextflow_pipeline {

    name "Test unzip"
    script "./main.nf"
    tag "unzip"

    test("unzip") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          ZipUtilsWrapper.zip("${params.outdir}/stable/stable_content.txt", "${params.outdir}/stable_content.zip")
          ZipUtilsWrapper.unzip("${params.outdir}/stable_content.zip", "${params.outdir}/stable_content_unzipped/")
          def stable_name = getAllFilesFromDir(params.outdir, true, [''], null, ['**'])
          assert snapshot(
            stable_name
          ).match()
        }
    }
}

