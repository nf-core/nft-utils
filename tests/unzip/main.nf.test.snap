{
    "unzip": {
        "content": [
            [
                [
                    "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d"
                ],
                "stable_content.txt:md5,f6d73f703cda1c725ea78369a6c3a48d",
                "stable_content.zip:md5,ba7d71d2c8e3cd2a47fa1b498da44e99",
                [

                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-16T17:57:07.994203269"
    }
}
